src/components/transport_manager/src/transport_adapter/transport_adapter_impl.cc

```diff
@@ -352,6 +363,7 @@ TransportAdapter::Error TransportAdapterImpl::DisconnectDevice(
   Error error = OK;
   DeviceSptr device = FindDevice(device_id);
   device->set_connection_status(ConnectionStatus::CLOSING);
+  ConnectionStatusUpdated();
```

Comment:
Might be able to integrate this into the `set_connection_status` function, need to look into.

Reply:
Resolved with `UpdateConnectionStatus`